api/user.go

```diff
@@ -623,13 +631,56 @@ func createProfileImage(username string, userId string) ([]byte, *model.AppError
 	h.Write([]byte(userId))
 	seed := h.Sum32()
 
+	initials := ""
+	parts := strings.Split(username, " ")
+
+	for _, v := range parts {
```

Comment:
I think this is old code based on the full name.  username cannot have spaces.  I would remove dead code.